### ISSUE TITLE
Calculation of individual labels evaluation performance

### ISSUE THREAD

**Issue by bundit786** [scikit-multilearn/scikit-multilearn#84]
Hi,
      Since using the equipped function from sklearn such as 'accuracy_score(y_test, predictions)' provides only the accuracy for the entire labels set, I would like to ask for function or method that can compute the evaluation performance for individual labels. I didn't see this stuff in scikit.ml tutorial.

Thanks and regards,
Bundit

**Comment by ljvmiranda921** [comment]
Hi @bundit786 , I think as of now there's no implementation for that. Although a naive solution would be to compare each column then return an array of values for each label. If you're interested to create a wrapper for accuracy_score then feel free to make a pull request! 👍

**Comment by ansin218** [comment]
I second @bundit786 here. Not only for accuracy, but also for other metrics such as precision, recall and F1. 

Also, @ljvmiranda921 could you please elaborate on that, may be I can give it a shot but as of now I am finding it too vague.

**Comment by ChristianSch** [comment]
[This](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) might already do what you're after.

If not, here are some thoughts. For a given (binary) label indiciator np-array, you just select the j-th column for label j (and do the same for your ground truth) and then calculate accuracy based on these two columns, which are of course just vectors in itself. That's how you'd get label-wise accuracy. Metrics that make sense for label-wise operations are:
* accuracy
* subset accuracy
* precision
* recall
* f1-score

Ranking losses/scores and  f1  in its aggregated micro/macro form does not make sense imho.

**Comment by ansin218** [comment]
@ChristianSch , I got the F1, precision and recall scores label wise based on the link. However, I am still struggling with how to get accuracy. Could you elaborate a bit, I could not quite comprehend your advise.

**Comment by ansin218** [comment]
Okay, I got this done. I will post the code shortly. :)

**Comment by ansin218** [comment]
Here is the code as promised. 

```
import pandas as pd
import numpy as np
import scipy
import scipy.sparse as sp
from sklearn.model_selection import train_test_split
from skmultilearn.problem_transform import BinaryRelevance
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.metrics import classification_report
from sklearn.cross_validation import cross_val_predict

data = pd.read_csv("your_csv_file.csv", dtype={'sentence':np.str_ })

y = data[['isA','isB','isC','isD','isE']]
to_drop = ['id','isA','isB','isC','isD','isE']
X = data.drop(to_drop, axis=1)

count_vectorizer = CountVectorizer()
counts = count_vectorizer.fit_transform(X['sentence'].values)

X_train, X_test, y_train, y_test = train_test_split(counts, y, test_size=0.33)

clf = BinaryRelevance(RandomForestClassifier())

clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)

# Print invidiual label wise F1, precision and recall scores
target_names = ['A', 'B', 'C', 'D', 'E']
print(classification_report(y_pred, y_test))

# Extract the values of label A alone, repeat the same for other labels
extractedData1 = y_pred[:,[1]]
z = scipy.sparse.csr_matrix(y_test)
extractedData2 = z[:,[1]]

# Flatten both the array and print the accuracy score for label A
actual = extractedData1.toarray().flatten()
predicted = extractedData2.toarray().flatten()
print(accuracy_score(actual, predicted))
```

**Note: For accuracy score, I have only computed for label A. Repeat the procedure for other labels!**

**Comment by niedakh** [comment]
```python 
>>> import sklearn.metrics as skm
>>> from skmultilearn.utils import measure_per_label
>>> measure_per_label(skm.accuracy_score, y_test, y_predicted)
[0.90199335548172754,
 0.96843853820598003,
 0.94352159468438535,
 0.93189368770764125,
 0.76910299003322258,
 0.84883720930232553]
```
